cluster-stamp.bicep

```diff
@@ -0,0 +1,1842 @@
+resource miKubeletManagedIdentityOperatorRole_roleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = {
+  scope: podmiIngressController
+  name: guid(resourceGroup().id, 'podmi-ingress-controller', managedIdentityOperatorRole)
+  properties: {
+    roleDefinitionId: managedIdentityOperatorRole
+    principalId: mc.properties.identityProfile.kubeletidentity.objectId
+    principalType: 'ServicePrincipal'
+  }
+  dependsOn: []
+}
+
+resource mcAadAdminGroupClusterAdminRole_roleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = if (isUsingAzureRBACasKubernetesRBAC) {
+  scope: mc
+  name: guid('aad-admin-group', mc.id, clusterAdminAadGroupObjectId)
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${clusterAdminRoleId}'
+    description: 'Members of this group are cluster admins of this cluster.'
+    principalId: clusterAdminAadGroupObjectId
+    principalType: 'Group'
+  }
+  dependsOn: []
+}
+
+resource mcAadAdminGroupServiceClusterUserRole_roleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = if (isUsingAzureRBACasKubernetesRBAC) {
+  scope: mc
+  name: guid('aad-admin-group-sc', mc.id, clusterAdminAadGroupObjectId)
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${serviceClusterUserRoleId}'
+    description: 'Members of this group are cluster users of this cluster.'
+    principalId: clusterAdminAadGroupObjectId
+    principalType: 'Group'
+  }
+  dependsOn: []
+}
+
+resource maAadA0008ReaderGroupClusterReaderRole_roleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = if (isUsingAzureRBACasKubernetesRBAC && (!(a0008NamespaceReaderAadGroupObjectId == clusterAdminAadGroupObjectId))) {
+  scope: nsA0008
+  name: guid('aad-a0008-reader-group', mc.id, a0008NamespaceReaderAadGroupObjectId)
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${clusterReaderRoleId}'
+    principalId: a0008NamespaceReaderAadGroupObjectId
+    description: 'Members of this group are cluster admins of the a0008 namespace in this cluster.'
+    principalType: 'Group'
+  }
+  dependsOn: []
+}
+
+resource maAadA0008ReaderGroupServiceClusterUserRole_roleAssignment 'Microsoft.Authorization/roleAssignments@2020-10-01-preview' = if (isUsingAzureRBACasKubernetesRBAC && (!(a0008NamespaceReaderAadGroupObjectId == clusterAdminAadGroupObjectId))) {
+  scope: mc
+  name: guid('aad-a0008-reader-group-sc', mc.id, a0008NamespaceReaderAadGroupObjectId)
+  properties: {
+    roleDefinitionId: '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${serviceClusterUserRoleId}'
```

Comment:
For these `roleDefinitionId` items (here and above).  One of two options, either use the bicep discouraged `subscriptionResourceId()` function or pull these subscription role definitions in as a `existing` resource and then `def.id` here -- with the latter being the most preferred.  Very very very few times should we be concating resource IDs like this.  Sorry I missed this on the first pass.

Reply:
sure thing @ckittel 

not even partially addressing what is being asked here, but sharing c0dac32f88bc0b318fb97ecd09710b003ba23a0e to spot other cases where we are concatenating, and discuss whether for a `var` it is acceptable, or we want to modify them all.

Reply:
I'm okay with this for now. I see we have other situations where this could have been applied as well.  So let's just say it's good for now!